hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SystemTransactions.java

```diff
@@ -391,6 +402,64 @@ ctx, createFileID(filesConfig.hapiPermissions(), config), bytes),
         autoNodeAdminKeyUpdates.tryIfPresent(adminConfig.upgradeSysFilesLoc(), systemContext);
     }
 
+    /**
+     * Cleans up council-controlled system accounts that are no longer needed.
+     * @param now the current time
+     * @param state the state to clean up
+     * @return true if the cleanup was finished, false otherwise
+     */
+    public boolean do065SystemAccountCleanup(@NonNull final Instant now, @NonNull final State state) {
+        requireNonNull(state);
+        requireNonNull(now);
+        final AtomicReference<AccountID> legacyAccountId = new AtomicReference<>();
+        // System context for dispatching CryptoDelete with an onSuccess callback
+        // that completely removes the legacy account from state after the dispatch
+        final var systemContext = newSystemContext(
+                now,
+                state,
+                dispatch -> {
+                    if (streamMode != RECORDS) {
+                        immediateStateChangeListener.resetKvStateChanges(null);
+                    }
+                    // On success, actually remove the legacy account from state
+                    final var tokenStates = state.getWritableStates(TokenService.NAME);
+                    final var accountsState = tokenStates.<AccountID, Account>get(ACCOUNTS_KEY);
+                    accountsState.remove(legacyAccountId.get());
+                    ((CommittableWritableStates) tokenStates).commit();
+                    if (streamMode != RECORDS) {
+                        final var changes = immediateStateChangeListener.getKvStateChanges();
+                        if (!changes.isEmpty()) {
+                            dispatch.streamBuilder().stateChanges(changes);
+                        }
+                    }
+                    // And decrement the entity count for the account type
+                    final var entityStates = state.getWritableStates(EntityIdService.NAME);
+                    final var entityCounters = new WritableEntityIdStore(entityStates);
+                    entityCounters.adjustEntityCount(ACCOUNT, -1);
+                    ((CommittableWritableStates) entityStates).commit();
```

Comment:
With the `.commit()` to the entity state, isn't a `dispatch.streamBuilder().stateChanges(...)` needed here too? (Like ~10 lines above)

Reply:
Megamap automatically defers singleton state changes to the end of the round, so this change to the `EntityCounts` singleton won't show up here anyways.